### PR TITLE
Revise/Collapse Checklist & HTML Refactor

### PR DESCRIPTION
**Pull request by jperson1**
# Revise/Collapse Checklist & HTML Refactor

Issue: https://github.com/GSA-TTS/FAC/issues/3161

## Changes:
1. After locking for certification, the checklist looks pretty different.
    * The form and upload pages are hidden away.
    * There's new key info text explaining the lock/unlock and preview downloads. This includes text about email notifications.
    * There are button links to the downloads and the unlock page.
    * The unlock button is not available after submission has been completed
2. HTML refactor:
    * Many checklist components are now in their own files. Notably, the different key info sections have their own files.
    * The section block has moved into the same directory as the other checklist components. The template tag has been updated to match.
    * The grand and glorious `submission_checklist.html` is much smaller due to these changes, and you can see the important logic on _one screen_. How crazy is that?
    * A bunch of comment updates that are probably/maybe useful. 
3. Minor things:
    * The download buttons on the actual certification screens are gone, in favor of the ones in the checklist.
    * 403's that come from accessing a cert screen without access now show why. Instead of "bad role", we have "Bad Role. User is not the Auditee/or Certifying Official". 

## How to Test:
1. Switch to this branch and run normally.
2. Go through a submission.
3. Ensure that things appear the same, until after locking for certification. 
4. After locking, ensure that the new form of the checklist:
    * Has the new key info text
    * Has new download & unlock buttons
    * Has _functional_ download and unlock buttons
5. Step through to submission. Make sure nothing breaks. 

As a bonus, you can run Cypress to verify that these changes are non-breaking for the full-submission tests. 

## Screenshots:
New checklist form:
image

Post submission/acceptance, the unlock goes away:
image

403 display change:
image

Removed the download buttons from the cert pages:
image

## PR checklist: submitters

- [x]   Link to an issue if possible. If there’s no issue, describe what your branch does. Even if there is an issue, a brief description in the PR is still useful.
- [x]   List any special steps reviewers have to follow to test the PR. For example, adding a local environment variable, creating a local test file, etc.
- [ ]   For extra credit, submit a screen recording like [this one](https://github.com/GSA-TTS/FAC/pull/1821).
- [x]   Make sure you’ve merged `main` into your branch shortly before creating the PR. (You should also be merging `main` into your branch regularly during development.)
- [x]   Make sure you’ve accounted for any migrations. When you’re about to create the PR, bring up the application locally and then run `git status | grep migrations`. If there are any results, you probably need to add them to the branch for the PR. Your PR should have only **one** new migration file for each of the component apps, except in rare circumstances; you may need to delete some and re-run `python manage.py makemigrations` to reduce the number to one. (Also, unless in exceptional circumstances, your PR should not delete any migration files.)
- [x]   Make sure that whatever feature you’re adding has tests that cover the feature. This includes test coverage to make sure that the previous workflow still works, if applicable.
- [x]   Make sure the full-submission.cy.js [Cypress test](https://github.com/GSA-TTS/FAC/blob/main/docs/testing.md#end-to-end-testing) passes, if applicable.
- [x]   Do manual testing locally. Our tests are not good enough yet to allow us to skip this step. If that’s not applicable for some reason, check this box.
- [x]   Verify that no Git surgery was necessary, or, if it was necessary at any point, repeat the testing after it’s finished.
- [ ]   Once a PR is merged, keep an eye on it until it’s deployed to dev, and do enough testing on dev to verify that it deployed successfully, the feature works as expected, and the happy path for the broad feature area (such as submission) still works.

## PR checklist: reviewers

- [ ]   Pull the branch to your local environment and run `make docker-clean; make docker-first-run && docker compose up`; then run `docker compose exec web /bin/bash -c "python manage.py test"`
- [ ]   Manually test out the changes locally, or check this box to verify that it wasn’t applicable in this case.
- [ ]   Check that the PR has appropriate tests. Look out for changes in HTML/JS/JSON Schema logic that may need to be captured in Python tests even though the logic isn’t in Python.
- [ ]   Verify that no Git surgery is necessary at any point (such as during a merge party), or, if it was, repeat the testing after it’s finished.

The larger the PR, the stricter we should be about these points.

### CODE REVIEW

**Comment by lauraherring** [comment]
Looks good to me from a content-perspective.

**Review by lauraherring — COMMENTED** [review]
Content is good.

File: backend/audit/mixins.py [hunk 1]
```diff
@@ -65,7 +65,8 @@ def dispatch(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpRespo
                 eligible_users = [acc.user for acc in eligible_accesses]
 
                 raise CertificationPermissionDenied(
-                    "bad role", eligible_users=eligible_users
+                    "Bad Role. User is not the Auditee Certifying Official.",
```
Comment (tadhg-ohiggins) [inline backend/audit/mixins.py:68]: Can we change this to `Invalid role. User is not the Auditee Certifying Official.`?

File: backend/audit/mixins.py [hunk 2]
```diff
@@ -93,7 +94,8 @@ def dispatch(self, request: HttpRequest, *args: Any, **kwargs: Any) -> HttpRespo
                 )
                 eligible_users = [acc.user for acc in eligible_accesses]
                 raise CertificationPermissionDenied(
-                    "bad role", eligible_users=eligible_users
+                    "Bad Role. User is not the Auditor Certifying Official",
```
Comment (tadhg-ohiggins) [inline backend/audit/mixins.py:97]: As above, can we change `Bad Role` to `Invalid role`?

File: backend/audit/mixins.py (hunk 1 above)
Comment (tadhg-ohiggins) [inline backend/audit/mixins.py:68]:
```suggestion
                    "Invalid role. User is not the Auditee Certifying Official.",
```

File: backend/audit/mixins.py (hunk 2 above)
Comment (tadhg-ohiggins) [inline backend/audit/mixins.py:97]:
```suggestion
                    "Invalid role. User is not the Auditor Certifying Official",
```

**Review by tadhg-ohiggins — COMMENTED** [review]
I made minor copy change suggestions.

File: backend/audit/mixins.py (hunk 1 above)
Comment (jperson1) [reply backend/audit/mixins.py]: Sure thing!

**Review by tadhg-ohiggins — APPROVED** [review]

**Pull request merged**